cmake/modules/AddSwift.cmake

```diff
@@ -488,6 +488,18 @@ function(_add_variant_link_flags)
     endif()
   endif()
 
+  # Enable dead stripping. Portions of this logic were copied from llvm's
+  # `add_link_opts` function (which, perhaps, should have been used here in the
+  # first place, but at this point it's hard to say whether that's feasible).
+  #
+  # TODO: Evaluate/enable -f{function,data}-sections --gc-sections for bfd,
+  # gold, and lld.
+  if(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
+    # See rdar://48283130: This gives 6MB+ size reductions for swift and
+    # SourceKitService, and much larger size reductions for sil-opt etc.
+    list(APPEND result "-Wl,-dead_strip")
```

Comment:
`add_link_opts` only adds the option for a release build. It may be useful to avoid stripping in debug builds because stripping can hide linker errors caused by dependency cycles.
Or try to just delegate to `add_link_opts` ?

Reply:
I'll add a Debug check. It'll take extensive refactoring to pull in the definition of `add_link_opts` here (just including AddLLVM breaks the swift build due to conflicting variables).

Reply:
Hmm, I'm certainly no CMake expert but just want to point out that I've added a call to `add_link_opts` in here:
https://github.com/apple/swift/blob/34f8670d2a8d1ce05c6da32da8e484e543e8ec5e/tools/libSwiftSyntaxParser/CMakeLists.txt#L21
without issues, but maybe this particular file is problematic for it.

Reply:
Yeah, add_link_opts requires that a target exists (it sets properties on the target directly). Most of swift's cmake logic collects properties, and *then* creates a target.

Maybe we should just factor out the core logic in add_link_opts into a helper which returns a list (in llvm), then use that here?